### PR TITLE
Damping by pass applied to previous displacement

### PR DESCRIPTION
**Pull request by gaborkb**
### Purpose of this PR

https://jira.unity3d.com/browse/CMCL-118
Described in this forum post: https://forum.unity.com/threads/problem-with-cinemachine-collider.948316/

The problem is that collider's displacement is in world space, and when damping is on and the FreeLook rotates, it does not also rotate the offset that's damping itself away.

Solutions could be either: keep offset in camera-local space, or **apply state's damping bypass to the offset**.

### Testing status

[Explanation of what’s tested, how tested and existing or new automation tests. Can include manual testing by self and/or QA. Specify test plans. Rarely acceptable to have no testing.]

- [ ] Added an automated test
- [ ] Passed all automated tests
- [x] Manually tested 

### Documentation status

[Overview of how documentation is affected by this change. If there is no effect on documentation, explain why. Otherwise, state which sections are changed and why.]

- [x] Updated [CHANGELOG](https://keepachangelog.com/en/1.0.0/)
- [ ] Updated README (if applicable)
- [ ] Commented all public classes, properties, and methods
- [ ] Updated user documentation

### Samples status

- [ ] Re-packed Extras~/CinemachineExamples.unitypackage

### Technical risk

[Overall product level assessment of risk of change. Need technical risk & halo effect.]

### Comments to reviewers

[Info per person for what to focus on, or historical info to understand who have previously reviewed and coverage. Help them get context.]

### Package version

[Justification for updating either the patch, minor, or major version according to the [semantic versioning](https://semver.org/spec/v2.0.0.html) rules]

- [ ] Updated package version

### CODE REVIEW

File: Runtime/Behaviours/CinemachineCollider.cs [hunk 1]
```diff
@@ -325,7 +325,8 @@ public override float GetMaxDampTime()
                     {
                         Vector3 delta = displacement - extra.m_previousDisplacement;
                         delta = Damper.Damp(delta, damping, deltaTime);
-                        displacement = extra.m_previousDisplacement + delta;
+                        displacement = 
+                            Quaternion.Euler(state.PositionDampingBypass) * extra.m_previousDisplacement + delta;
```
Comment (glabute) [inline Runtime/Behaviours/CinemachineCollider.cs:329]: This looks fishy.  You use an unrotated prev displacement in the first line.  Shouldn't you rotate it first?

**Review by glabute — CHANGES_REQUESTED** [review]
It's better, but I don't think this fix does the whole job.  I can still get it to move laterally in this situation.  See the big horizontal displacement of the vcam (yellow line).  If I then rotate to an unobstructed place, there is slow horizontal damping back to normal

image

**Comment by gaborkb** [comment]
> It's better, but I don't think this fix does the whole job. I can still get it to move laterally in this situation. See the big horizontal displacement of the vcam (yellow line). If I then rotate to an unobstructed place, there is slow horizontal damping back to normal
> 
> image

I think, we should fix that in another PR. Because it think it will be a feature change, not a bug fix.

**Comment by glabute** [comment]
ok, we can do it in a second pass

**Review by glabute — APPROVED** [review]

**Pull request merged**